pkg/adapter/apiserver/adapter.go

```diff
@@ -75,15 +77,40 @@ func (a *apiServerAdapter) Start(stopCh <-chan struct{}) error {
 
 	a.logger.Infof("STARTING -- %#v", a.config)
 
-	for _, r := range a.config.Resources {
-		lw := &cache.ListWatch{
-			// TODO: this will not work with cluster scoped resources.
-			ListFunc:  asUnstructuredLister(a.k8s.Resource(r.GVR).Namespace(a.config.Namespace).List, r.LabelSelector),
-			WatchFunc: asUnstructuredWatcher(a.k8s.Resource(r.GVR).Namespace(a.config.Namespace).Watch, r.LabelSelector),
+	for _, configRes := range a.config.Resources {
+
+		resources, err := a.kubeClient.Discovery().ServerResourcesForGroupVersion(configRes.GVR.GroupVersion().String())
```

Comment:
I wonder if this needs more RBAC? I have not used this call before.

Reply:
Discovery API is fairly open.

`system:discovery` cluster role is granted to `system:authenticated` by default (an I think that's a grant that admins cannot revoke).